### PR TITLE
chore: support various IDEs and editors

### PR DESCRIPTION
**Pull request by kkyueing**
Add `.eslintrc` and `.editorconfig` to support consistent coding style across various IDEs and editors.

### CODE REVIEW

File: app/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+root = true
```
Comment (OHHAKO) [inline app/.editorconfig:1]: is it for activation?

File: app/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+[*]
```
Comment (OHHAKO) [inline app/.editorconfig:3]: what is this star?

File: app/.editorconfig (hunk 1 above)
Comment (kkyueing) [reply app/.editorconfig:1]: This statement declares that the current directory is the root directory of the project. This prevents editors and IDEs to scan parent directories to find other `.editorconfig` files.

Comment (OHHAKO) [reply app/.editorconfig:1]: i got it.

File: app/.editorconfig (hunk 2 above)
Comment (kkyueing) [reply app/.editorconfig:3]: Asterisk mark here means "all files," so instructions in this section will be applied for every files in the project. If you want to override some instructions for specific files, e.g. Javascript files, then you may append `[*.js]` section to the end of the file.

File: app/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "parser": "babel-eslint",
```
Comment (OHHAKO) [inline app/.eslintrc:2]: i think eslint is linting tool and also made by 'babel'

File: app/gatsby-config.js [hunk 4]
```diff
@@ -28,6 +28,7 @@ module.exports = {
         icon: `src/images/gatsby-icon.png`, // This path is relative to the root of the site.
       },
     },
+    `gatsby-plugin-eslint`,
```
Comment (OHHAKO) [inline app/gatsby-config.js:31]: i guess it's for import the eslint plugin

File: app/.editorconfig (hunk 2 above)
Comment (OHHAKO) [reply app/.editorconfig:3]: so comfortable mark it is! thank you 😄

**Pull request merged**